prow/pod-utils/gcs/target.go

```diff
@@ -57,17 +57,28 @@ func AliasForSpec(spec *pjutil.JobSpec) string {
 }
 
 // LatestBuildForSpec determines the GCS path for storing the latest
-// build id for a job.
-func LatestBuildForSpec(spec *pjutil.JobSpec) string {
+// build id for a job. pathSegment can be nil so callers of this
+// helper are not required to choose a path strategy but can still
```

Comment:
This is a bit smelly -- in what case are we calling this wihout a `repopathbuilder`?

Reply:
tot only needs a single path. Or we could hardcode a `repobuilder`? Or make it configurable in tot?

Reply:
`tot` needs to be configured the same way as the pushers, else it's going to look in the wrong place for these files?

Reply:
* `initupload` needs the `repobuilder` only for presubmits
* presubmit `latest-build.txt`s are pushed in two different places:
  - `pr-logs/directory/latest-build.txt`
  - `pr-logs/pull/org_repo/pull_number/job_name/latest-build.txt`
* tot only needs one path and the most updated sits in `pr-logs/directory/latest-build.txt`.

Reply:
Why do we push to the other one, then?

Reply:
Seems that Gubernator is using the PR-specific `latest-build.txt` @rhhm will know better since he requested it in https://github.com/kubernetes/test-infra/pull/7068#issuecomment-370061849

Reply:
@rmmh

Reply:
@rmmh what's the use case there?

Reply:
Yes, there's recent runsf or both.